Bump k8s to 1.14.1

ingvagabund opened the pull request:
- bump k8s to v1.14.1
- Manager.GetRecorder renamed to Manager.GetEventRecorderFor
- replace "-" with "_" in controller names (the name is used to prefix prometheus metric name which refuses "-" in metric name, see https://github.com/kubernetes-sigs/controller-runtime/pull/376)
  - machineset-controller -> machineset_controller
  - machinedeployment-controller -> machinedeployment_controller
  - cluster-controller -> cluster_controller
- replace order of arguments in Client.List (runtime object switched with list options, list options converted to a function)
- schema moved from "sigs.k8s.io/controller-runtime/pkg/runtime/schema" under "sigs.k8s.io/controller-runtime/pkg/scheme"
- "k8s.io/client-go/util/integer" moved under "k8s.io/utils/integer"

ingvagabund committed:
UPSTREAM: <carry>: openshift: Bump k8s to 1.14.1

Requires to update controller-runtime to the latest which has just recently adopted k8s v1.14.

ingvagabund commented:
/refresh
/retest

ingvagabund commented:
/retest

bison commented on pkg/controller/machinedeployment/machinedeployment_controller_test.go:
```diff
@@ -89,7 +89,7 @@ func TestReconcile(t *testing.T) {
 	machineSets := &machinev1beta1.MachineSetList{}
 	expectInt(t, 1, func(ctx context.Context) int {
-		if err := c.List(ctx, &client.ListOptions{}, machineSets); err != nil {
+		if err := c.List(ctx, machineSets, func(opts *client.ListOptions) {}); err != nil {
```
What's up with the empty functions? Can you not pass `nil`?

ingvagabund replied:
It panics unfortunately. Already tried.

bison replied:
What is the panic? That seems like a bug for sure. We should at least open an issue about that and link here or something.

bison replied:
Also, I guess you shouldn't even need `nil`. You should be able to just not pass this parameter.

bison commented on pkg/controller/machinedeployment/controller.go:
```diff
@@ -108,9 +108,7 @@ func add(mgr manager.Manager, r reconcile.Reconciler, mapFn handler.ToRequestsFu
-	listOptions := &client.ListOptions{Namespace: d.Namespace}
-
-	if err := r.Client.List(context.Background(), listOptions, machineSets); err != nil {
+	if err := r.Client.List(context.Background(), machineSets, func(opt *client.ListOptions) { opt.InNamespace(d.Namespace) }); err != nil {
```
This looks weird. Aren't the options here a variadic parameter, and they give you a bunch of functions that return `ListOptionFunc`, so you can do something like what's below?

`r.Client.List(context.Background(), machineSets, client.InNamespace(ns), client.MatchingLabels(labels))`

ingvagabund replied:
Good point. PR updated.

ingvagabund committed:
UPSTREAM: <carry>: openshift: regen clientset, bazel and CRDs

frobware commented on pkg/controller/machine/controller.go:
```diff
@@ -61,7 +61,7 @@ func AddWithActuator(mgr manager.Manager, actuator Actuator) error {
 	r := &ReconcileMachine{
 		Client:        mgr.GetClient(),
-		eventRecorder: mgr.GetRecorder("machine-controller"),
+		eventRecorder: mgr.GetEventRecorderFor("machine-controller"),
```
Should this also be s/-/_?

ingvagabund replied:
I would still prefer to keep the event source to have `-` instead of `_`.

ingvagabund committed:
UPSTREAM: <carry>: openshift: Alter code to accept updated sigs.k8s.io/controller-runtime@master

- Manager.GetRecorder renamed to Manager.GetEventRecorderFor
- replace "-" with "_" in controller names (the name is used to prefix prometheus metric name which refuses "-" in metric name)
  - machineset-controller -> machineset_controller
  - machinedeployment-controller -> machinedeployment_controller
  - cluster-controller -> cluster_controller
- replace order of arguments in Client.List (runtime object switched with list options, list options converted to a function)
- schema moved from "sigs.k8s.io/controller-runtime/pkg/runtime/schema" under "sigs.k8s.io/controller-runtime/pkg/scheme"
- "k8s.io/client-go/util/integer" moved under "k8s.io/utils/integer"

frobware commented:
/approve

frobware commented:
/lgtm
